pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/PinotLogicalQueryPlanner.java

```diff
@@ -51,70 +48,72 @@ private PinotLogicalQueryPlanner() {
-      subPlanRootSenderNode.addInput(subPlanRoot);
-      PlanFragment planFragment1 = new PlanFragment(1, subPlanRootSenderNode, new ArrayList<>());
-      planFragmentMap.put(1, planFragment1);
-      for (Int2ObjectMap.Entry<IntList> entry : childPlanFragmentIdsMap.int2ObjectEntrySet()) {
-        PlanFragment planFragment = planFragmentMap.get(entry.getIntKey());
-        List<PlanFragment> childPlanFragments = planFragment.getChildren();
-        IntListIterator childPlanFragmentIdIterator = entry.getValue().iterator();
-        while (childPlanFragmentIdIterator.hasNext()) {
-          childPlanFragments.add(planFragmentMap.get(childPlanFragmentIdIterator.nextInt()));
-        }
-      }
-      MailboxReceiveNode rootReceiveNode =
-          new MailboxReceiveNode(0, subPlanRoot.getDataSchema(), subPlanRoot.getPlanFragmentId(),
-              RelDistribution.Type.BROADCAST_DISTRIBUTED, PinotRelExchangeType.getDefaultExchangeType(), null, null,
-              false, false, subPlanRootSenderNode);
-      PlanFragment rootPlanFragment = new PlanFragment(0, rootReceiveNode, Collections.singletonList(planFragment1));
-      SubPlan subPlan = new SubPlan(rootPlanFragment, subPlanContext._subPlanIdToMetadataMap.get(0), new ArrayList<>());
-      subPlanMap.put(subPlanId, subPlan);
-    }
-    for (Map.Entry<Integer, List<Integer>> subPlanToChildrenEntry : subPlanContext._subPlanIdToChildrenMap.entrySet()) {
-      int subPlanId = subPlanToChildrenEntry.getKey();
-      List<Integer> subPlanChildren = subPlanToChildrenEntry.getValue();
-      for (int subPlanChild : subPlanChildren) {
-        subPlanMap.get(subPlanId).getChildren().add(subPlanMap.get(subPlanChild));
-      }
-    }
-    return subPlanMap.get(0);
+//    SubPlanFragmenter.Context subPlanContext = new SubPlanFragmenter.Context();
+//    subPlanContext._subPlanIdToRootNodeMap.put(0, rootNode);
+//    subPlanContext._subPlanIdToMetadataMap.put(0,
+//        new SubPlanMetadata(RelToPlanNodeConverter.getTableNamesFromRelRoot(relRoot.rel), relRoot.fields));
+//    rootNode.visit(SubPlanFragmenter.INSTANCE, subPlanContext);
+//
+//    Map<Integer, SubPlan> subPlanMap = new HashMap<>();
+//    for (Map.Entry<Integer, PlanNode> subPlanEntry : subPlanContext._subPlanIdToRootNodeMap.entrySet()) {
+//      SubPlan subPlan =
+//          new SubPlan(planNodeToPlanFragment(subPlanEntry.getValue()), subPlanContext._subPlanIdToMetadataMap.get(0),
+//              new ArrayList<>());
+//      subPlanMap.put(subPlanEntry.getKey(), subPlan);
+//    }
+//    for (Map.Entry<Integer, List<Integer>> subPlanToChildrenEntry : subPlanContext._subPlanIdToChildrenMap.entrySet
+//    ()) {
+//      int subPlanId = subPlanToChildrenEntry.getKey();
+//      List<Integer> subPlanChildren = subPlanToChildrenEntry.getValue();
+//      for (int subPlanChild : subPlanChildren) {
+//        subPlanMap.get(subPlanId).getChildren().add(subPlanMap.get(subPlanChild));
+//      }
+//    }
+//    return subPlanMap.get(0);
```

Comment:
Shouldn't we remove this code?

Reply:
Intentionally commented it out and added a TODO so that we can revisit in the future